docs/tutorials/Fine-Tuning-Semantic-Parsing.md

```diff
@@ -0,0 +1,352 @@
+        "composite_score": composite_score,
+    }
+```
+
+### Results
+
+We evaluated the fine-tuned models on a hold-out evaluation set of 714 examples and computed the metrics described above.
+The results are shown in the table below.
+
+| Metric | Llama2-7B Chat | Full-weight Fine-tuning | LoRA Fine-tuning |
+| ------------------------ | ----- | ----- | ----- |
+| Function Name Accuracy   | 0.014 | 0.577 | 0.517 |
+| Attribute Set Accuracy   | 0.010 | 0.822 | 0.845 |
+| Attribute Value Accuracy | 0.524 | 0.942 | 0.881 |
+| Overall Accuracy         | 0.183 | 0.780 | 0.748 |
+
+There are a few highlights from the results:
+
+- The baseline Llama2-7B Chat model's performance is remarkably low at 0.183 overall accuracy. This is consistent with our earlier observation that it does not really understand how to perform the task.
+- Fine-tuning methods, both full-weight and LoRA, substantially enhance the model's accuracy, achieving 0.780 and 0.748, respectively. Notably, LoRA fine-tuning approaches the metrics of full-weight fine-tuning and achieves a better `Attribute Set Accuracy`, while training with less than 0.1% parameters. This highlights the efficiency of LoRA.
+- Full-weight fine-tuning outperforms LoRA fine-tuning in the `Function Name Accuracy` and the `Attribute Value Accuracy` metrics. This shows the advantage of training the entire model weights on the task. Though further hyperparameter tuning might allow LoRA to close this gap.
+- The higher accuracy in attribute set and value suggests that the attributes are more contextually driven and thus easier for the model to predict. In contrast, correctly identifying function names appears to be more challenging, indicating a need for deeper understanding of the task and reasoning capability.
+
+#### Confusion among Function Names
+
+To expand on the last point, we went deep into predictions of function name.
+The confusion matrices below illustrate how the the full-weight fine-tuning and LoRA fine-tuning models perform in this area.
+
+![Confusion Matrix of Full-weight Fine-Tuning](../figures/finetune_func_cm_full_weight.png)
+![Confusion Matrix of LoRA Fine-Tuning](../figures/finetune_func_cm_lora.png)
+
+- A notable area of confusion is within the `confirm` column, where both models frequently misclassify examples from the `inform`, `recommend`, and `suggest` categories as `confirm`. This may stem from these function names being too similar in the dataset, leading to ambiguity that the models struggle to differentiate.
+- The full-weight fine-tuning model (depicted on the left) tends to make more errors in the "empty" or "other" columns, indicating instances where the model either fails to predict a function name or predicts one that doesn't exist within the predefined set.
+- The LoRA fine-tuning model (shown on the right) exhibits more confusion between the predefined function classes; overall, it has fewer correct predictions along the diagonal of the matrix.
```

Comment:
this is weird to me because it goes against the narrative in my head that LoRA is good for teaching the structure of a task but bad for injecting new information. Do you have any thoughts?

Reply:
Yes, the amount of confusion is indeed, bit high. 

I think the results on attributes fields and value does show the advantage of fine-tuning at "teaching about form/structure": both models outputs attributes in the correct format and contains the correct information. 

What I want to show here is that, the selection of the function names is more ambiguous and difficult. It is beyond the "form" problem, but more on reasoning, and the difference between functions `confirm`, `inform`, `suggest` and `recommend` does not seem clear. So I think reducing # functions, clarify the difference, and possibly improve label quality, would be key to improve on the function name accuracy.